server/index.js

```diff
@@ -1,9 +1,53 @@
-const express = require('express')
-const app = express()
-const port = 3000
+const express = require('express');
+const app = express();
+const port = 3000;
+const db = require('../database/index.js');
+const parser = require('./parser.js');
 
 app.use(express.static(__dirname + '/../client/dist'));
 
+app.get('/community/:projectId', (req, res) => {
+  let projectId = req.params.projectId, cityIds = [], countryChance = [], cityChance = [], countryNames = [], cityNames = [], finalLocations;
+
+  Promise.resolve(db.getLocations(projectId))
+    .then(projectLocations => {
+
+      if (projectLocations.length === 0) {
+        throw new Error('Invalid project id. The provided project id is out of range.');
+      } else {
+        projectLocations = JSON.parse(projectLocations[0].locations)
+      }
+
+      let countries = Object.keys(projectLocations);
+
+      for (let country in projectLocations) {
+        let cities = projectLocations[country].cities;
+        countryChance.push(projectLocations[country].chance);
+
+        cities.forEach(city => {
+          cityIds.push(Object.keys(city)[0]);
```

Comment:
Does the order of city's keys matter? You're pushing the first key, but given that city is an object that value may change. Same with cityChance below.